testsuites/syncgateway/functional/tests/test_log_rotation.py

```diff
@@ -0,0 +1,466 @@
+import json
+import os
+
+import pytest
+
+from keywords.ClusterKeywords import ClusterKeywords
+from keywords.RemoteExecutor import RemoteExecutor
+from keywords.SyncGateway import SyncGateway
+from keywords.SyncGateway import sync_gateway_config_path_for_mode
+from keywords.utils import log_info
+from libraries.testkit.cluster import Cluster
+
+
+
+@pytest.mark.sanity
+@pytest.mark.syncgateway
+@pytest.mark.parametrize("sg_conf_name", [
+    "custom_sync/sync_gateway_log_rotation"
+])
+def test_log_logKeys_string(params_from_base_test_setup, sg_conf_name):
+    cluster_conf = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    log_info("Running 'sync_access_sanity'")
+    log_info("Using cluster_conf: {}".format(cluster_conf))
+    log_info("Using sg_conf: {}".format(sg_conf))
+
+    cluster = Cluster(config=cluster_conf)
+    cluster.reset(sg_config_path=sg_conf)
+
+    with open(sg_conf) as defaul_conf:
+        data = json.load(defaul_conf)
+
+    data['logging']["default"]["logKeys"] = "http"
+
+    temp_conf = "/".join(sg_conf.split('/')[:-2]) + '/temp_conf.json'
+
+    with open(temp_conf, 'w') as fp:
+        json.dump(data, fp)
+
+    # Stop sync_gateways
+    log_info(">>> Stopping sync_gateway")
+    sg_helper = SyncGateway()
+    cluster_helper = ClusterKeywords()
+    cluster_hosts = cluster_helper.get_cluster_topology(os.environ["CLUSTER_CONFIG"])
+    sg_one_url = cluster_hosts["sync_gateways"][0]["public"]
+    sg_helper.stop_sync_gateway(cluster_config=os.environ["CLUSTER_CONFIG"], url=sg_one_url)
+    try:
+        sg_helper.start_sync_gateway(cluster_config=os.environ["CLUSTER_CONFIG"], url=sg_one_url, config=temp_conf)
```

Comment:
If I understand it correctly, SG should not start with logKeys set to http and should start normally if it is set to * ? Can you add comments?